### ISSUE TITLE
请教单机训练和集群训练模型auc差别非常大

### ISSUE THREAD

**Issue by jinyongbo128** [PaddlePaddle/Paddle#7923]

**Comment by jinyongbo128** [comment]
model:dssm

net_conf:
user_feature_layer = paddle.layer.data("user_fea", paddle.data_type.sparse_binary_vector(user_dict_dim))
query_feature_layer = paddle.layer.data("query_fea",paddle.data_type.sparse_binary_vector(query_dict_dim))
lab = paddle.layer.data("score", type=paddle.data_type.dense_vector(1))

hidden_0_0 = paddle.layer.fc(input=query_feature_layer, size=64, act=paddle.activation.Relu(), param_attr=paddle.attr.ParamAttr(sparse_update=True, l1_rate=0.01))
hidden_0_1 = paddle.layer.fc(input=hidden_0_0, size=10, act=paddle.activation.Tanh())

hidden_1_0 = paddle.layer.fc(input=user_feature_layer, size=64, act=paddle.activation.Relu(), param_attr=paddle.attr.ParamAttr(sparse_update=True, l1_rate=0.01))
hidden_1_1 = paddle.layer.fc(input=hidden_1_0, size=50, act=paddle.activation.Relu())
hidden_1_2 = paddle.layer.fc(input=hidden_1_1, size=10, act=paddle.activation.Tanh())

output1 = paddle.layer.cos_sim(a=hidden_1_2, b=hidden_0_1, size=1)
output = paddle.layer.fc(input=output1, size=1, act=paddle.activation.Sigmoid())
cost = paddle.layer.multi_binary_label_cross_entropy_cost(input=output, label=lab)

learn_optimizer = paddle.optimizer.AdaGrad(learning_rate=1e-3, regularization=paddle.optimizer.L2Regularization(rate=1e-3))

现象是：同一个train_data，同一predict_data，单机版训练模型auc=0.75，但是集群跑出的model，auc=0.51...
翻看以往issue，目前尝试过如下调参方式，均没得打改善：
1. batch_size 分别调整过100,20, 1，都没有改善
2. node_num原先为100，调整AdaGrad的learning_rate为learning_rate/node_num, L2正则rate调为rate*node_num

请问单机和集群差异这么大的原因，会有哪些，还有其他方式调优吗？

**Comment by chengduoZH** [comment]
@jinyongbo128  您是说learning_rate和L2正则rate调整前后对模型准确率没有影响，是吗？

**Comment by jinyongbo128** [comment]
我现在情况是单机训练auc=0.75，集群训练auc就只有0.5。。这结果肯定有问题
我调整learning_rate和L2正则rate，试图想让集群训练的auc结果更好，但目前看没有改善

**Comment by jinyongbo128** [comment]
@chengduoZH 补充说明下，是同样的训练和测试数据，分别用于单机和集群训练，两者训练完后模型效果差异非常大，集群的模型auc=0.5，单机的auc=0.75

**Comment by typhoonzero** [comment]
1. 可以检查下`trainer = paddle.trainer.SGD(...., is_local=False)`这个`is_local=False`在集群训练中需要加上否则就会各训各的，不会merge模型
1. `learning_rate/node_num`这个在v1/v2集群训练必须加。
1. "试图想让集群训练的auc结果更好"： 集群训练是在训练速度和效果上的一个取舍，用多个节点只可以让同样的时间可以训更多的数据，而多数情况，单机训练的效果往往会高于集群训练。

**Comment by jinyongbo128** [comment]
@typhoonzero 
1. is_local=False  这个之前训练都加了的
2. learning_rate/node_num  尝试加过，但是效果不是很明显。现在主要是auc=0.5，等于完全随机，感觉本身哪里就有问题。。我现在小数据集train，train的inst数只有50w，这个情况batch_size多少合适？

**Comment by newdimitri** [comment]
您好，
想请问PaddlePaddle支持哪几种分布式模式呢？

**Comment by typhoonzero** [comment]
@newdimitri paddle fluid从设计来讲可以支持数据并行和模型并行。目前的版本主要支持数据并行模式。并可以支持同步训练和异步训练，以及nccl2 同步训练，详细参考：http://www.paddlepaddle.org/documentation/docs/zh/0.15.0/user_guides/howto/training/multi_node.html

**Comment by feitonlife** [comment]
我这边也是遇到了同样的问题，单机auc稳定上升，paddlecloud＋厂内mpi集群，fluid0.15，auc始终0.5
